context/app/static/js/components/cells/DatasetsSelectedByExpression/hooks.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useCallback, useState } from 'react';
+import CellsService from 'js/components/cells/CellsService';
+import { useAppContext } from 'js/components/Contexts';
+import { fetchSearchData } from 'js/hooks/useSearchData';
+import useCellsChartLoadingStore, { CellsChartLoadingStore } from 'js/stores/useCellsChartLoadingStore';
+import { useStore, CellsSearchStore } from 'js/components/cells/store';
+import { useAccordionStep } from 'js/shared-styles/accordions/StepAccordion';
+import { useSnackbarActions } from 'js/shared-styles/snackbars';
+
+const chartsStoreSelector = (state: CellsChartLoadingStore) => state.resetFetchedUUIDs;
+
+const cellsStoreSelector = (state: CellsSearchStore) => ({
+  setResults: state.setResults,
+  minExpressionLog: state.minExpressionLog,
+  setMinExpressionLog: state.setMinExpressionLog,
+  minCellPercentage: state.minCellPercentage,
+  setMinCellPercentage: state.setMinCellPercentage,
+  cellVariableNames: state.cellVariableNames,
+  setCellVariableNames: state.setCellVariableNames,
+  queryType: state.queryType,
+  setIsLoading: state.setIsLoading,
+});
+
+function getSearchQuery(cellsResults: { uuid: string }[]) {
```

Comment:
I didn't really touch the code for the `getSearchQuery` function besides adding type annotations, but if there's an opportunity to remove this in favor of an already-existing function then we can take it. Could we go over this in more detail since you've had more recent experience with the ES queries?